### DISCUSSION TITLE
Memory Issues When Running Next.js on Shared Hosting

### DISCUSSION THREAD

**Discussion by ulinnuhadeni** [vercel/next.js#67617] [Help]
### Summary

I have created static next.js project (without fetching external APIs) on my shared hosting cpanel based with specifications :

- RAM : 2 GB
- vCPU: 2
- Node Version : 20.14.0
- Next.js Version : 14.2.4

i tried to build locally and manual upload into server ran well for first time. 
then i asked some people (around 10 people) to access it, 

and ram usage increasing from 102 MB (initial access after restart project) to 1,1 GB. 
so  i restarted the next.js application and ram came back to 102 MB

is there another approach to avoid ram increasing when accessing project ?

**Comment by BorhanSaflo** [comment]
It sounds like it's a memory leak. To start, make sure all your dependencies are up to date, including Next.js, as updates often come with bug fixes that might solve your problem. 

If that doesn't work, you'd have to dig a bit deeper by using memory profiling tools. Take a look at the [Record a Heap Profile](https://nextjs.org/docs/app/building-your-application/optimizing/memory-usage#record-a-heap-profile) section in the docs. By analyzing these profiles, you can see what objects are eating up memory.

**Reply by ulinnuhadeni** [reply]
thanks for advise, 

i have found the solution,
with put cron job task to kill npm every hour.

i ve tried with that approach seems my memory not going increase over and over
